components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -1541,6 +1541,33 @@ public static String getGatewayendpoint(String transports) {
         return null;
     }
 
+    /**
+     * Read the GateWay Endpoint from the APIConfiguration. If multiple Gateway
+     * environments defined, get the gateway endpoint according to the environment type
+     *
+     * @param transports      transports allowed for gateway endpoint
+     * @param environmentType gateway environment type
+     * @return Gateway URL
+     */
+    public static String getGatewayEndpoint(String transports, String environmentName, String environmentType) {
+        String gatewayURLs;
+        String gatewayEndpoint = "";
+
+        Map<String, Environment> gatewayEnvironments = ServiceReferenceHolder.getInstance()
```

Comment:
What is the key of this map? Can't we simply do map.get(environmentName)

Reply:
the key seems to be the gateway name.